cassandra/4.0/modules/ROOT/pages/cassandra-connector-examples.adoc

```diff
@@ -0,0 +1,575 @@
+
+If you don’t know how to use a properties file, see Configuring Property Placeholders.
+
+=== Configure a Template for the HTML Input Form
+
+This flow configures an HTML template to use for providing HTML input to create the new table:
+
+.Flow for configuring an HTML template
+image::cassandra-parse-template.png[Parse Template Flow]
+
+==== Configure HTTP Listener
+
+Configure HTTP Listener to initiate a Mule flow when a call is made to the `/` path on port 8081:
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign (*+*) next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In *Listener* configuration window, set the *Path* field value to `/`.
+
+==== Add a Parse Template Component
+
+Add a *Parse Template* component to create a template for an HTML form. The example uses this form to convey information for creating the table. (*Reviewers, is this text correct?*):
+
+. From the Mule Palette view, search for `parse` and drag *Core* > *Parse Template* to the right of *Listener*.
+. In the *Location* field, enter `form.html`.
+*Reviewers, where is the information for this template? I assume it is in form.html, but where is the form put?*
+
+=== Create the Keyspace
+
+This flow creates the keyspace for the new table:
+
+.Flow for creating a keyspace
+image::cassandra-create-keyspace.png[Create keyspace flow]
+
+==== Configure HTTP listener
+
+Configure HTTP Listener to initiate a Mule flow when a call is made to the `/test/createKeyspace` path on port `8081`:
+
+. Create a new Mule Project in Anypoint Studio.
+. In Studio, click *HTTP* and drag the *Listener* operation to the canvas.
+. On the Listener properties window, click the plus sign *+* next to the *Connector configuration* field to add a global element.
+. Accept the defaults.
+. In *Listener* configuration window, set the *Path* field value to `/test/createKeyspace`.
+
+==== Add a Transform Message Component
+
+Add a *Transform Message* component to set Java variables based on the payload received from *HTTP Listener*: *Reviewers, is this description correct*?
+
+. From the Mule Palette view, drag a *Transform Message* component to the right of *Listener*.
+. In the *Transform Message* configuration, overlay the brackets in the *Output* section with this XML:
++
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/java
+---
+{
+ "keyspaceName": payload.keyspaceName,
+ "replicationFactor": payload.replicationFactor,
+ "replicationStrategyClass": payload.replicationStrategyClass
+}
+as Object {
+ class: "org.mule.modules.cassandradb.api.CreateKeyspaceInput"
+}
+----
+
+==== Add the Create Keyspace Operation
+
+Add the *Create keyspace* operation to create the new keyspace for the new table: (*Reviewers: please correct and make this text better*)
```

Comment:
Create a keyspace for the new table to exist in. The tables are created in a keyspace, so you need to create a keyspace first.